sdk/export/metric/metric.go

```diff
@@ -105,6 +107,10 @@ type AggregationSelector interface {
 // to attach a Sum aggregator to a ValueRecorder instrument or a
 // MinMaxSumCount aggregator to a Counter instrument.
 type Aggregator interface {
+	// Aggregation indicates the kind of Aggregation supported by
+	// this Aggregator.
+	aggregation.Aggregation
```

Comment:
I don't know about embedding the `aggregation.Aggregation` interface here. Maybe instead just an explicit method? E.g.

```
AggregationKind() aggregation.Kind
```

Or something similar.

Embedding the `aggregation.Aggregation` interface here means that an `Aggregator` implements an `Aggregation` and can be returned as such, which is not what we want (right?).

Having a method instead would better decouple the packages and prevent future refactors to split these apart if the Aggregation interface needs to be extended, but not extended here. It also makes the intent much clearer. Instead of explaining in the comment that the embedded interface is only there to explain the `Kind` the function signature does that job.

Additionally, it is a bit confusing when the `aggregation.Aggregation` interface is defined as ...

> ... an interface returned by the Aggregator containing an interval of metric data.

But it is embedded in the the thing that produces it based on this definition.

Reply:
You're right. I've changed it to a method named `Aggregation()`, which makes it explicit when an Aggregator is turned into an Aggregation. I wanted to say that this was always possible, not to make it happen implicitly.  If the Aggregator only supported an AggregationKind() accessor, the Integrator code would be forced to use a dynamic type check, which this approach avoids.